core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/ChecksumCalculatingAsyncRequestBody.java

```diff
@@ -215,6 +216,14 @@ private ByteBuffer getFinalChecksumAppendedChunk(ByteBuffer byteBuffer) {
             return checksumAppendedBuffer;
         }
 
+        private ByteBuffer appendChunkSizeAndFinalByte(ByteBuffer byteBuffer) {
+            ByteBuffer contentChunk = createChunk(byteBuffer, false);
+            ByteBuffer checksumAppendedBuffer = ByteBuffer.allocate(contentChunk.remaining());
+            checksumAppendedBuffer.put(contentChunk);
+            checksumAppendedBuffer.flip();
```

Comment:
I don't understand the need for this. Why create a copy of `contentChunk`?

Reply:
We need to  need to separately create `contentChunk` because we need to add following
- prefix = chunk size in hex + CRLF
- Suffix = CRLF

Reply:
Removed